plugins/camera/camera_impl.cpp

```diff
@@ -0,0 +1,154 @@
+#include "camera_impl.h"
+#include "device.h"
+#include "global_include.h"
+#include "mavlink_include.h"
+#include <functional>
+
+namespace dronecore {
+
+CameraImpl::CameraImpl(Device &device) :
+    PluginImplBase(device)
+{
+    _parent.register_plugin(this);
+}
+
+CameraImpl::~CameraImpl()
+{
+    _parent.unregister_plugin(this);
+}
+
+void CameraImpl::init() {}
+
+void CameraImpl::deinit() {}
+
+void CameraImpl::enable() {}
+
+void CameraImpl::disable() {}
+
+Camera::Result CameraImpl::take_photo()
+{
+    return camera_result_from_command_result(
+               _parent.send_command_with_ack(
+                   MAV_CMD_IMAGE_START_CAPTURE,
+                   MavlinkCommands::Params {0.0f, // all camera IDs
+                                            0.0f, // only picture, no interval
+                                            1.0f, // take only one picture
+                                            float(_capture_sequence++),
+                                            NAN, NAN, NAN},
+                   MAV_COMP_ID_CAMERA));
```

Comment:
As we discussed, there'll be one instance per camera; so, camera component ID would be a member. Isn't it ?

Reply:
That's right. For now I'm just going to add the current single device plugin. Later we can extend it.